### PR TITLE
Add single host test scripts to run in PRs

### PR DESCRIPTION
**Pull request by flacatus**
This PR create scripts to run single host deployment and workspace creation.
Refference issue: https://github.com/eclipse/che/issues/17944
Signed-off-by: Flavius Lacatusu <flacatus@redhat.com>

### CODE REVIEW

File: .ci/cico_openshift_shost.sh [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+  if [ $x -gt 180 ]
+  then
+    echo "[ERROR] Workspace didn't start after 3 minutes."
+    exit 1
+  fi
+}
+
+# Utility to get che events and pod logs from openshift cluster
+function getOCCheClusterLogs() {
```
Comment (tolusha) [inline .ci/cico_openshift_shost.sh:90]: can we use `chectl server:logs` to grab all logs and events ?

Comment (flacatus) [reply .ci/cico_openshift_shost.sh:90]: I'll investigate. I thing even `chectl server:logs`. All pod logs will be saved in openshift ci artifacts

File: .ci/cico_openshift_shost.sh [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+# Deploy Eclipse Che in single host mode
+function run() {
+    # Patch file to pass to chectl
+    cat >/tmp/che-cr-patch.yaml <<EOL
+spec:
+  server:
+    serverExposureStrategy: 'single-host'
+    singleHostExposureType: 'gateway'
```
Comment (tolusha) [inline .ci/cico_openshift_shost.sh:114]: This is my mistake. 
This line it is not needed at all for OpenShift.
Remove pls

File: .ci/cico_minikube_shost.sh [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+init
+source "${OPERATOR_REPO}"/.ci/util/ci_common.sh
+echo "[INFO] Start to Building Che Operator Image"
+buildCheOperatorImage
+
+echo "[INFO] Start to run single host with gateway exposure mode"
+runSHostGatewayExposure
+
+echo "[INFO] Start to run single host in native mode"
+runSHostNativeExposure
```
Comment (tolusha) [inline .ci/cico_minikube_shost.sh:197]: Two different checks is better then a single one

File: .github/workflows/single-host-tests.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#    available under the terms of the Eclipse Public License 2.0
+#    which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+#  SPDX-License-Identifier: EPL-2.0
+#
+#  Contributors:
+#    Red Hat, Inc. - initial API and implementation
+#
+
+name: Testing single host nightly deployment
```
Comment (tolusha) [inline .github/workflows/single-host-tests.yaml:13]: Let's name it:
`Testing latest changes [single-host/gateway]` and `Testing latest changes [single-host/native]`
Maybe somebody have better ideas.

File: .ci/cico_minikube_shost.sh [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+# exit immediately when a command fails
+set -e
+# only exit with zero if all commands of the pipeline exit successfully
+set -o pipefail
+# error on unset variables
+set -u
+# print each command before executing it
+set -x
+
+#Stop execution on any error
```
Comment (AndrienkoAleksandr) [inline .ci/cico_minikube_shost.sh:22]: Minor: `#Stop` -> `# Stop`

File: .ci/cico_minikube_shost.sh [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+function init() {
+  export SCRIPT=$(readlink -f "$0")
+  export SCRIPT_DIR=$(dirname "$SCRIPT")
+  export OPERATOR_REPO=$(dirname "$SCRIPT_DIR");
+
+  export RAM_MEMORY=8192
+  export NAMESPACE="che"
+  export PLATFORM="kubernetes"
+
+  # Directory where che artifacts will be stored and uploded to GH actions artifacts
```
Comment (AndrienkoAleksandr) [inline .ci/cico_minikube_shost.sh:35]: uploded -> uploaded

File: .ci/cico_minikube_shost.sh (hunk 3 above)
Comment (flacatus) [reply .ci/cico_minikube_shost.sh:188]: Understood. I'll change!

File: .ci/cico_openshift_shost.sh [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+EOL
+    echo "======= Che cr patch ======="
+    cat /tmp/che-cr-patch.yaml
+
+    # Start to deploy Che
+    chectl server:start --platform=openshift --skip-kubernetes-health-check --installer=operator \
+      --chenamespace=${NAMESPACE} --che-operator-cr-patch-yaml=/tmp/che-cr-patch.yaml --che-operator-image=${OPERATOR_IMAGE}
+
+    # Create and start a workspace
+    getSingleHostToken # Function from ./util/ci_common.sh
```
Comment (AndrienkoAleksandr) [inline .ci/cico_openshift_shost.sh:116]: > # Function from ./util/ci_common.sh

Is this comment actual? For me looks like you used this function from the cico_openshift_shost.sh

Comment (flacatus) [reply .ci/cico_openshift_shost.sh:116]: I'm changing things right now. I would like to start to diferentiate openshift ci script with github action scripts

**Review by tolusha — APPROVED** [review]

**Pull request merged**